ui/src/data-products/element-list/element-table.tsx

```diff
@@ -0,0 +1,124 @@
+                                )}
+                            </th>
+                            <th>
+                                {t(
+                                    'data-products.element-list.element-list-table.track-address-end',
+                                )}
+                            </th>
+                            <th>
+                                {t(
+                                    'data-products.element-list.element-list-table.coordinate-system',
+                                )}
+                            </th>
+                            <th>
+                                {t(
+                                    'data-products.element-list.element-list-table.location-start-e',
+                                )}
+                            </th>
+                            <th>
+                                {t(
+                                    'data-products.element-list.element-list-table.location-start-n',
+                                )}
+                            </th>
+                            <th>
+                                {t('data-products.element-list.element-list-table.location-end-e')}
+                            </th>
+                            <th>
+                                {t('data-products.element-list.element-list-table.location-end-n')}
+                            </th>
+                            <th>{t('data-products.element-list.element-list-table.length')}</th>
+                            <th>
+                                {t(
+                                    'data-products.element-list.element-list-table.curve-radius-start',
+                                )}
+                            </th>
+                            <th>
+                                {t(
+                                    'data-products.element-list.element-list-table.curve-radius-end',
+                                )}
+                            </th>
+                            <th>{t('data-products.element-list.element-list-table.cant-start')}</th>
+                            <th>{t('data-products.element-list.element-list-table.cant-end')}</th>
+                            <th>
+                                {t('data-products.element-list.element-list-table.angle-start')}
+                            </th>
+                            <th>{t('data-products.element-list.element-list-table.angle-end')}</th>
+                            <th>{t('data-products.element-list.element-list-table.plan')}</th>
+                            <th>{t('data-products.element-list.element-list-table.source')}</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {plans.map((item, index) => (
+                            // Element list can contain the same element multiple times -> use index in list as key
+                            <React.Fragment key={`${index}`}>
```

Comment:
Keyksi kannattaa melkein aina asettaa jokin muu kuin indeksi. React käyttää key:tä DOM-elementtien identifioimiseen, täältä löytyy lisää tietoa: https://reactjs.org/docs/lists-and-keys.html#keys

Reply:
Koitin keksiä tähän jotain fiksua datapohjaista indeksiä, mutta ongelmana on se, että jatkuvan geometrian haussa sama elementti voi tulla listalle useaan kertaan -> mikään ainakaan puhtaasti datapohjainen indeksi ei toimi. Tilanne voi (tietääkseni, correct me if I'm wrong) siis olla seuraavan kaltainen:

```
Plan 1:
AAAAAAAAAAAAAAAAAAAA
Plan 2: 
------BBBBBB------

Linkitykset paikannuspohjasta:
AAAAAABBBBBBAAAAAA
```

Tällöin tuo elementti A näytetään muistaakseni taulukossa kahteen kertaan

Reply:
Voisiko tuossa toimia esim keynä elementId + lengthMeters tai elementId + start.coordinate.x/y?
Eikö tuossa käy nyt hieman huonosti, jos element-tablen saama elements-propsun lista muuttuu?

Reply:
Jutellaas tästä vaikka dailyn jälkeen, sillä en oo itekään enää ihan satavarma miten tää menee 🤔 Listan indeksi toimii kumminkin Reactin näkökulmasta ihan hyvin indeksinä, ja vaikka se ei ideaalinen olekaan, niin ainakaan collisioneita ei pääse tapahtumaan